### DISCUSSION TITLE
Lensflare plugin

### DISCUSSION THREAD

**Discussion by Elius94** [mistic100/Photo-Sphere-Viewer#832] [Show and tell]
https://www.npmjs.com/package/photo-sphere-viewer-lensflare-plugin

# photo-sphere-viewer-lensflare-plugin
 Plugin to add lens flares to a Photo Sphere Viewer (Library: [PhotoSphereViewer](https://github.com/mistic100/Photo-Sphere-Viewer)) by [mistic100](https://github.com/mistic100)

 ![lensflare](https://user-images.githubusercontent.com/14907987/212742055-8db7599e-2e93-4303-b979-9762f803483d.gif)

### NPM

```bash
npm install photo-sphere-viewer-lensflare-plugin
```

### Yarn

```bash
yarn add photo-sphere-viewer-lensflare-plugin
```

### HTML

```html
<script src="https://cdn.jsdelivr.net/npm/photo-sphere-viewer-lensflare-plugin/index.min.js"></script>
```

### JavaScript

```js
import PhotoSphereViewerLensflarePlugin from 'photo-sphere-viewer-lensflare-plugin';
```

```js
const viewer = new PhotoSphereViewer.Viewer({
    container: document.querySelector('#viewer'),
    panorama: 'pano.jpg',
    plugins: [
        [PhotoSphereViewerLensflarePlugin, {
            lensflares: [
                {
                    id: 'sun',
                    position: { yaw: '145deg', pitch: '2deg' },
                    type: 0,
                }
            ]
        }]
    ]
});
```

**Comment by mistic100** [comment]
Okay that's awesome! Just tested it on my demo panorama, it works so well.  
Happy to see a first third party plugin released.

Like I did for the frameworks, I can add a "Third party plugins" page.

----

A few remarks:

Do you really need all the events ? It looks like you copied MarkersPlugin with a big search and replace, but I think you forgot to cleanup (like `LeaveLensflareEvent`, `EnterLensflareEvent`, etc).

You should remove `__checkObjectsObserver` as you don't use intersection detection on your ThreeJs objects, it consumes ressources doing rays casting for nothing.

**Reply by Elius94** [reply]
Thank you for appreciation!
You are right! I based myself on the structure of the Markers.
Tomorrow I should release version 1.0.0, which will include the complete cleanup, and if I can, some customization.
In the meantime, I'm glad you liked it ;)
I would like it to be included in the documentation, as you already mentioned. Thank you

**Reply by mistic100** [reply]
https://photo-sphere-viewer.js.org/plugins/third-party.html
